sdk/metric/pipeline.go

```diff
@@ -349,7 +349,8 @@ func (i *inserter[N]) cachedAggregator(scope instrumentation.Scope, kind Instrum
 	normID := id.normalize()
 	cv := i.aggregators.Lookup(normID, func() aggVal[N] {
 		b := aggregate.Builder[N]{
-			Temporality: i.pipeline.reader.temporality(kind),
+			Temporality:      i.pipeline.reader.temporality(kind),
+			AggregationLimit: cardinalityLimit(),
```

Comment:
Eventually, this will be gathered from the stream because it's a config option on a view.  How will we tell unset, e.g. use the default, from someone who wants to disable the feature?

Reply:
When the cardinality limit comes from a view, we will need to define the cardinality limit on the `Stream` in a way that we can distinguish unset from zero. I expect we can use something similar to [`Extrema`](https://github.com/open-telemetry/opentelemetry-go/blob/d37d851bbce65577ab340849ec6c540fdc6e3096/sdk/metric/metricdata/data.go#L208-L223).

As for the environment variable configuration, unset (or `""`) vs set will be the way we distinguish the default vs set.

As for disabling cardinality limits entirely, that is something yet to be well defined in the specification. There is prior thought on the concept in the original PR discussions:

- https://github.com/open-telemetry/opentelemetry-specification/pull/2960#discussion_r1034412798

However, the ability to completely disable cardinality limits was not resolved.

I think defaulting to "disabled" is adequate for the experimental phase of this feature. Prior to stabilizing this feature (here and in the specification) the "disable" question will need to be answered.
To disable the feature, the user will need to set a value of `-1`